Sources/ScipioKit/DescriptionPackage.swift

```diff
@@ -149,7 +149,7 @@ struct DescriptionPackage {
 }
 
 extension DescriptionPackage {
-    func resolveBuildProducts() throws -> [BuildProduct] {
+    func resolveBuildProducts() throws -> OrderedSet<BuildProduct> {
```

Comment:
@dmhts You wished to avoid using `OrderedSet`, but I think it's better to use it here.
https://github.com/giginet/Scipio/issues/130

What do you think?

Reply:
My concern about using `OrderedSet` was caused but not understanding at the time the real reason for the crash explained in https://github.com/giginet/Scipio/issues/130, which was the violation of the `Hashable` requirements in the first place. 
The reason for that should be fixed along with this PR now when we've implemented correct custom conformance to `Hashable` and `Equatable`. I'm not sure if `OrderedSet` is even needed here. 

Of course, it wouldn't hurt, but wouldn't `resolveBuildProducts` always produce an array of unique build products anyway? I think `topologicalSort` always outputs an array of `[Hashable]` without duplicates. 
Or do you use `OrderedSet` here just for some extra peace of mind here?

Reply:
Oh, As you said, I confirmed it's enough to add the `==` override to fix this error without modifying it here.

`OrderedSet` may have issues. It causes runtime crash 🤔 
However, I prefer to use this to clarify it doesn't have duplicated products if we can use it.

Reply:
> OrderedSet may have issues. It causes runtime crash

That's right, but it's important to note that it might happen only in the debug builds (to catch early certain performance-related issues). So I wouldn't be worried about the crash in the first place.

> However, I prefer to use this to clarify it doesn't have duplicated products if we can use it.

I believe using `OrderedSet` can be theoretically harmful, by preventing the identification of some subtle potential issues in the `resolveBuildProducts` implementation in the future. Like it happened just right now, with `OrderedSet` being used you would've never spotted the issue.

Reply:
As an alternative you can still convert the output to `OrderSet` but at least add before that an `assert` that would check that the output from the topological sort contains no duplicates.

Reply:
Thank you. If we use `Set` and runtime assertion here, it's difficult to write tests for this, because the order will be missing on the test case.

I'll try to use `OrderSet,` but I understand your concern about using it, so I'll keep paying attention.

Reply:
> If we use Set and runtime assertion here, it's difficult to write tests for this

Nope, I meant `OrderedSet` + the runtime assertion. An `assertion` is also a form of a unit test if you will. Something like `assert(Set(products).count < products.count)` could catch any issue with duplicates for debug builds and would run along with the main unit tests.

But you decide of course.